backend/src/v4/models/scene.js

```diff
@@ -75,12 +75,25 @@ Scene.findNodesByType = async function (account, model, branch, revision, type,
 	return findNodes(account, model, branch, revision, query, projection);
 };
 
+Scene.findMetadataNodesByFields = async function (account, model, branch, revision, fieldNames, projection = {}) {
+	const history = await History.getHistory(account, model, branch, revision);
+	const query = { $match: { rev_id: history._id, type: "meta" } };
+	projection = { $project: { ...projection, metadata: 1, parents: 1 } };
+
+	if (fieldNames.length) {
```

Comment:
I think it's better to assume the user may not pass in `[]` if they don't want to filter. 

```suggestion
	if (fieldNames?.length) {
```

Or you do `fieldNames=[]` on L78